Server/user/service.ts

```diff
@@ -27,23 +26,23 @@ export const userSignUp = async (req: Request, res: Response) : Promise<void> =>
 
         const {email, password} : {email : string , password : string} = req.body;
 
-        await User.find({"email" : email}).exec(async(err, foundUser) => {
-
-            if(foundUser.length !== 0) return res.status(403).send("Account is already exist");
-            else {
-                
-                const newUser = new User({
-                    email,
-                    password : getEncryptedPassword(password)
-                });
-                
-                await newUser.save();
-                return res.status(200).send("User added successfully");
-            }
-        });
+        const foundUser = await User.findOne({"email" : email});
+
+        if(foundUser) res.status(403).send("Account is already exist");
```

Comment:
Maybe be better to return res.status(.... everywhere. This could improve code. In this case: 
```
     if (foundUser) return res.status(403).send("Account is already exist")
      const newUser = new User({
          email,
          password : getEncryptedPassword(password)
     });
     ..........
```
P.S. Without ```else```